src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,93 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        putValue(hash(key), key, value);
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> newNode = table[hash(key) % table.length];
+        while (newNode != null) {
+            if (Objects.equals(key, newNode.key)) {
+                return newNode.value;
+            }
+            newNode = newNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putValue(int hash, K key, V value) {
```

Comment:
```suggestion
    private void putValue(K key, V value) {
```
let's not pass this variable if we can calculate it from key, that will improve code readability